.github/workflows/test.yml

```diff
@@ -5,21 +5,44 @@ on:
     types: [opened, synchronize, reopened, ready_for_review]
 
 jobs:
+  smoke-test:
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-latest
+    steps:
+    -
+      name: Checkout repository
+      uses: actions/checkout@v2
+    -
+      name: Run Smoke Tests
+      run: |
+        # Ensure scripts in repository are executable
+        IFS=$'\n';
+        for f in $(find . -name '*.sh'); do if [[ ! -x $f ]]; then echo "$f is not executable" && FAIL=1; fi ;done
+        unset IFS;
+        # If FAIL is 1 then we fail.
+        [[ $FAIL == 1 ]] && exit 1 || echo "Smoke Tests Passed"
```

Comment:
I'm a bit irritated by the `||` (OR).
But I guess it reads

```
If [[ $FAIL == 1 ]]; then
  exit 1
else
  echo "Smoke Tests Passed"
fi
```

Reply:
Yep, that's exactly what it says. I've not experimented with multi-line statements in a `run` section yet, but didn't really need to as this works just as well.